File: sdks/fog/challenges/all_connections.rb
#!/usr/bin/env ruby

require 'fog'

connection_opts = {
    :rackspace_username   => ENV['RAX_USERNAME'],
    :rackspace_api_key    => ENV['RAX_API_KEY'],
    :rackspace_region     => ENV['RAX_REGION'].downcase.to_sym,
    :rackspace_auth_url   => "#{ENV['RAX_AUTH_URL']}/v2.0"
}

# Compute services:
compute = Fog::Compute.new(connection_opts.merge({:provider => 'rackspace'}))
puts "Servers: #{compute.servers.all}"
puts "Networks: #{compute.networks.all}"

load_balancers = Fog::Rackspace::LoadBalancers.new(connection_opts)
puts "Cloud Load Balancers: #{load_balancers.load_balancers.all}"

storage = Fog::Storage.new(connection_opts.merge({:provider => 'rackspace'}))
puts "Cloud Files Containers: #{storage.directories.all}"

databases = Fog::Rackspace::Databases.new(connection_opts)
puts "Cloud Databases: #{databases.instances.all}"

dns = Fog::DNS::Rackspace.new(connection_opts)
puts "Cloud DNS: #{dns.zones.all}"

identity = Fog::Rackspace::Identity.new(connection_opts)
puts "Cloud Identity Users: #{identity.users.all}"

monitoring = Fog::Rackspace::Monitoring.new(connection_opts)
puts "Cloud Monitoring Account: #{monitoring.list_entities}"

block_storage = Fog::Rackspace::BlockStorage.new(connection_opts)
puts "Cloud Block Storage Volumes: #{block_storage.volumes.all}"

# Cloud Backup?

autoscale = Fog::Rackspace::AutoScale.new(connection_opts)
puts "Autoscale Scaling Groups: #{autoscale.groups.all}"

# Cloud Queues
queues = Fog::Rackspace::Queues.new(connection_opts)
puts "Cloud Queues: #{queues.queues.all}"

File: sdks/php-opencloud/challenges/all_connections.php
<?php
require 'vendor/autoload.php';
use OpenCloud\Rackspace;

$region = getenv('RAX_REGION');
$endpoint = getenv('RAX_AUTH_URL') . '/v2.0/';
$credentials = array(
    'username' => getenv('RAX_USERNAME'),
    'apiKey' => getenv('RAX_API_KEY')
);

$rackspace = new Rackspace($endpoint, $credentials);
$rackspace->Authenticate();

$compute = $rackspace->computeService('cloudServersOpenStack', $region);
$servers = $compute->serverList();
$networks = $compute->networkList();
echo("Servers: ".var_export($servers));
echo("Networks: ".var_export($networks));

$loadBalancerService = $rackspace->loadBalancerService('cloudLoadBalancers', $region);
$load_balancers = $loadBalancerService->loadBalancerList();
echo("Cloud Load Balancers: ".var_export($load_balancers));

$cloudFilesService = $rackspace->objectStoreService('cloudFiles', $region);
$containers = $cloudFilesService->listContainers();
echo("Cloud Files Containers: ".var_export($containers));

$databaseService = $rackspace->databaseService('cloudDatabases', $region);
$databases = $databaseService->instanceList();
echo("Cloud Databases: ".var_export($databases));

$dnsService = $rackspace->dnsService('cloudDNS', $region);
$domains = $dnsService->domainList();
// Can I get zones?
// echo("Cloud DNS: $zones")
echo('Cloud DNS: $domains');

$identityService = $rackspace->identityService();
$users = $identityService->getUsers();
echo("Cloud Identity Users: ".var_export($users));

$monitoringService = $rackspace->cloudMonitoringService('cloudMonitoring', 'ORD', 'publicURL');
$checkTypes = $monitoringService->getCheckTypes();
// Can I get the account info?
// echo("Cloud Monitoring Account: $account")
echo("Cloud Monitoring Check Types: ".var_export($checkTypes));

$blockStorageService = $rackspace->volumeService('cloudBlockStorage', 'DFW');
$volumes = $blockStorageService->volumeList(false);
echo("Cloud Block Storage Volumes: ".var_export($volumes));

# Cloud Backup?

$autoscaleService = $rackspace->autoscaleService();
$groups = $autoscaleService->groupList();
echo("Autoscale Scaling Groups: ".var_export($groups));

# Cloud Queues
$queuesService = $rackspace->queuesService('cloudQueues', $region);
$queues = $queuesService->listQueues();
echo("Cloud Queues: ".var_export($queues));

?>
